sys/net/gnrc/network_layer/sixlowpan/gnrc_sixlowpan.c

```diff
@@ -293,12 +293,36 @@ static void _send(gnrc_pktsnip_t *pkt)
         return;
     }
 
-#ifdef MODULE_GNRC_SIXLOWPAN_IPHC
-    if (netif->flags & GNRC_NETIF_FLAGS_6LO_HC) {
-        gnrc_sixlowpan_iphc_send(pkt, NULL, 0);
+    if (IS_USED(MODULE_GNRC_SIXLOWPAN_IPHC) &&
+        netif->flags & GNRC_NETIF_FLAGS_6LO_HC) {
+        gnrc_sixlowpan_frag_fb_t *fbuf;
+
+        if (IS_USED(MODULE_GNRC_SIXLOWPAN_FRAG_HINT) &&
+            IS_USED(MODULE_GNRC_SIXLOWPAN_FRAG_MINFWD)) {
+            /* prepare for sending with IPHC slack in first fragment */
+            fbuf = gnrc_sixlowpan_frag_fb_get();
+            if (fbuf != NULL) {
+                fbuf->pkt = pkt;
+                fbuf->datagram_size = datagram_size;
+                fbuf->tag = gnrc_sixlowpan_frag_fb_next_tag();
+                fbuf->offset = 0;
+#if IS_USED(MODULE_GNRC_SIXLOWPAN_FRAG_HINT)
```

Comment:
This is already checked by the encapsulating if block (line 300)

Reply:
True, but the field is only available with `IS_USED(MODULE_GNRC_SIXLOWPAN_FRAG_HINT)` so either we use CPP `#if` for the whole block or just isolated here (or otherwise will get a compiler error, when this module is missing)

Reply:
Will add a comment